reflex-dom-core/src/Reflex/Dom/Builder/Immediate.hs

```diff
@@ -162,13 +164,22 @@ instance MonadJSM m => MonadJSM (ImmediateDomBuilderT t m) where
     liftJSM' = ImmediateDomBuilderT . liftJSM'
 #endif
 
-data ImmediateDomBuilderEnv t
-   = ImmediateDomBuilderEnv { _immediateDomBuilderEnv_document :: Document
-                            , _immediateDomBuilderEnv_parent :: Node
-                            , _immediateDomBuilderEnv_unreadyChildren :: IORef Word -- Number of children who still aren't fully rendered
-                            , _immediateDomBuilderEnv_commitAction :: JSM () -- Action to take when all children are ready --TODO: we should probably get rid of this once we invoke it
-                            }
+-- |Structure forming the environment in which a DOM builder runs, tracking what document, where in that document is being built, and so on.
+data ImmediateDomBuilderEnv t = ImmediateDomBuilderEnv
+  { _immediateDomBuilderEnv_document :: Document
+  -- ^The document in which building is being performed, kept separately nodes can be created without having to obtain the owner document from the parent node.
```

Comment:
"... kept separately _so that_ notes can be created ...", i think.

Might also want to add a TODO to determine whether this is actually a good idea.  I believe the original reason was that `ownerDocument` wasn't available in ghcjs-dom, or something like that - not performance, specifically.